### DISCUSSION TITLE
Disable column movement on focus

### DISCUSSION THREAD

**Discussion by Minion3665** [YaLTeR/niri#490] [Ideas]
So, I have multiple monitors, and sometimes I want to do things like leaving a terminal and a YouTube video on one monitor, and moving to another.

The problem is, sometimes when I do that, I have a window partially on (or just off) the screen, and this causes niri to focus that window and shift everything around... because of this, I'd like to stop niri from moving my columns, even if I focus one that's not on the screen.

I found the [`center-focused-column`](https://github.com/YaLTeR/niri/wiki/Configuration:-Layout#center-focused-column) option, and thought it was what I needed, but unfortunately setting `"never"` for that still scrolls to the edge of the screen rather than preventing movement altogether. I couldn't find a config setting that did exactly what I wanted.

One mitigation could be disabling [`focus-follows-mouse`](https://github.com/YaLTeR/niri/wiki/Configuration:-Input#focus-follows-mouse), but I like `focus-follows-mouse` and don't really want to disable it for this.

**Comment by Minion3665** [comment]
This is pretty close to #484, but it's not quite the same -- for example that one talks about only fullscreen windows whereas for me even moving firefox onto the screen fully here if I focused it would constitute unwanted movement. Also, that discussion seems to be talking about denying focus, whereas I'm fine to focus a window that's partially offscreen, and interact with it as normal despite not being able to see all of it.
![image](https://github.com/YaLTeR/niri/assets/34243578/0b3ec265-57eb-42e8-bdb9-4064fd207db8)

It's possible my idea would mitigate the problem in that discussion, but I'm not entirely sure and didn't want to take over that -- @IgnIVertiKalCaD what do you think of this idea? Would it help to have this posted in your discussion too/instead?

**Reply by IVKLD** [reply]
All previously said, voiced my desires dyal niri. I would like to place 3 windows on one screen, but if I move the cursor to the right side of the monitor, it does not fly to the window that is on the right. I would like it to only move when I either click on the right side of the monitor or with the keyboard.

I'd like a "position anchoring" feature. That is, under no circumstances change the position of my viewport (I am only familiar with front-end development and can only use terms from there).

For ponitia - Hyprland. Those who have worked on this tiling will understand. (Yes, it doesn't have horizontal scrolling, but it's a good example).

**Comment by YaLTeR** [comment]
Let's discuss in https://github.com/YaLTeR/niri/discussions/491
